mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -2716,6 +2719,55 @@ impl OrdermatchContext {
     }
 }
 
+#[derive(Default)]
+struct MakerOrdersContext {
+    orders: HashMap<Uuid, Arc<AsyncMutex<MakerOrder>>>,
+    count_by_tickers: HashMap<String, usize>,
+    balance_loops: HashMap<String, AbortOnDropHandle>,
+}
+
+impl MakerOrdersContext {
+    fn add_order(&mut self, ctx: MmWeak, order: MakerOrder, balance: Option<BigDecimal>) {
+        if !self.balance_loop_exists(&order.base) {
+            let ticker = order.base.to_string();
+            let handle = spawn_abortable(async move { check_balance_update_loop(ctx, ticker, balance).await });
+            self.add_balance_loop(&order.base, handle);
+        }
+
+        *self.count_by_tickers.entry(order.base.clone()).or_insert(0) += 1;
+        self.orders.insert(order.uuid, Arc::new(AsyncMutex::new(order)));
+    }
+
+    fn remove_order(&mut self, uuid: &Uuid, ticker: &str) -> Option<Arc<AsyncMutex<MakerOrder>>> {
```

Comment:
The function user can pass incorrect `ticker` (for example, he passes `rel` coin instead of `base`).
Also, taking the `ticker` argument makes it difficult to use the function in some places.

The best way to refactor the function is to remove the order from `MakerOrdersContext::orders` and get the `MakerOrder::base` ticker from the removed item. But since all items wrapped into `AsyncMutex`, it'll require to make the function async and to lock the mutex (that can lead to race conditions).

Another way that I prefer is to add `MakerOrdersContext::order_tickers: HashMap<Uuid, String>`.
And then this function will look like:
```rust
fn remove_order(&mut self, uuid: &Uuid) -> Option<Arc<AsyncMutex<MakerOrder>>> {
        let order = self.orders.remove(uuid)?;
        let ticker = self.order_tickers.remove(uuid)?;
        if let Some(count) = self.count_by_tickers.get_mut(&ticker) {
            if *count > 0 {
                *count -= 1;
            }
        }

        if !self.coin_has_active_maker_orders(&ticker) {
            self.remove_balance_loop(&ticker);
        }
        Some(order)
    }
```
Of course you need to push the ticker to `MakerOrdersContext::order_tickers` on `MakerOrdersContext::add_order`.